source/expressions.tex

```diff
@@ -1477,12 +1490,24 @@
 \begin{codeblock}
 void f() {
   float x, &r = x;
-  [=] {
+  [=]() -> decltype((x)) {      // lambda returns \tcode{float const\&} because this lambda is not \tcode{mutable} and
+                                // \tcode{x} is an lvalue
     decltype(x) y1;             // \tcode{y1} has type \tcode{float}
-    decltype((x)) y2 = y1;      // \tcode{y2} has type \tcode{float const\&} because this lambda
-                                // is not \keyword{mutable} and \tcode{x} is an lvalue
+    decltype((x)) y2 = y1;      // \tcode{y2} has type \tcode{float const\&}
     decltype(r) r1 = y1;        // \tcode{r1} has type \tcode{float\&}
     decltype((r)) r2 = y2;      // \tcode{r2} has type \tcode{float const\&}
+    return y2;
+  };
+
+  [=]<decltype(x) P>{};         // error: \tcode{x} refers to local entity but precedes the
+                                // lambda's function parameter scope
+  [=](decltype((x)) y){};       // error: \tcode{x} refers to local entity but is in the lambda's
+                                // \grammarterm{parameter-declaration-clause}
+  [=]{
+      []<decltype(x) P>{};      // OK: \tcode{x} is in the outer lambda's function parameter scope
```

Comment:
@jensmaurer: should we use "OK, " here, too?

Reply:
Yes. This probably needs global cleanup.
```suggestion
      []<decltype(x) P>{};      // OK, \tcode{x} is in the outer lambda's function parameter scope
```